### PR TITLE
GAWB-2838: BigQuery inside notebooks

### PR DESCRIPTION
**Pull request by jmthibault79**
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/project/Dependencies.scala [hunk 1]
```diff
@@ -38,6 +38,7 @@ object Dependencies {
     "org.scalatest"       %%  "scalatest"     % "2.2.6"   % "test",
     "org.seleniumhq.selenium" % "selenium-java" % "3.8.1" % "test",
     "com.typesafe.scala-logging" %% "scala-logging" % "3.1.0",
+    "org.apache.commons" % "commons-text" % "1.2",
```
Comment (jmthibault79) [inline automation/project/Dependencies.scala]: for `StringEscapeUtils`

File: project/Dependencies.scala [hunk 2]
```diff
@@ -11,7 +11,7 @@ object Dependencies {
 
   val workbenchUtilV    = "0.2-d97f551"
   val workbenchModelV   = "0.9-7a4d3bd"
-  val workbenchGoogleV  = "0.10-b95b2c1"
+  val workbenchGoogleV  = "0.11-7ad0aa8"
```
Comment (jmthibault79) [inline project/Dependencies.scala]: For BigQuery scopes

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala [hunk 3]
```diff
@@ -77,7 +78,8 @@ class NotebookPage(override val url: String)(override implicit val authToken: Au
   def executeCell(code: String, timeoutSeconds: Long = 60): Option[String] = {
     await enabled cells
     val cell = lastCell
-    executeScript(s"""arguments[0].CodeMirror.setValue("$code");""", cell)
+    val jsEscapedCode = StringEscapeUtils.escapeEcmaScript(code)
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookPage.scala]: Nice, thanks

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala [hunk 4]
```diff
@@ -460,5 +468,36 @@ class LeonardoSpec extends FreeSpec with Matchers with Eventually with ParallelT
...
+        // project owners have the bigquery role automatically, so this also tests granting it to users
+
+        val ronEmail = LeonardoConfig.Users.ron.email
+        Orchestration.billing.addUserToBillingProject(projectName.value, ronEmail, Orchestration.billing.BillingProjectRole.User)(ownerToken)
+        Orchestration.billing.addGoogleRoleToBillingProjectUser(projectName.value, ronEmail, "bigquery.jobUser")(ownerToken)
+
+        implicit val leoToken: AuthToken = ronAuthToken
+        withNewCluster(projectName) { cluster =>
+          withNewNotebook(cluster) { notebookPage =>
+            val query = """! bq query "SELECT COUNT(*) AS scullion_count FROM publicdata.samples.shakespeare WHERE word='scullion'" """
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala]: https://www.opensourceshakespeare.org/views/plays/play_view.php?WorkID=henry4p2&Act=2&Scene=1&Scope=scene

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala]: I also considered rampallian and fustilarian.

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala [hunk 5]
```diff
@@ -460,5 +468,36 @@ class LeonardoSpec extends FreeSpec with Matchers with Eventually with ParallelT
...
+            val expectedResult =
+              """+----------------+
+                || scullion_count |
+                |+----------------+
+                ||              2 |
+                |+----------------+""".stripMargin
```
Comment (helgridly) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala]: Adding `--format=json` to your `bq` command will probably make this less ridiculous.
https://cloud.google.com/bigquery/bq-command-line-tool#usefulflags

**Comment by jmthibault79** [comment]
Swat passes -> https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/732/

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala (hunk 5 above)
Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoSpec.scala]: ok, done

**Pull request merged**